### ISSUE TITLE
ScheduleReduction cannot reduce into a single number

### ISSUE THREAD

**Issue by shmsong** [csarofeen/pytorch#491]
## 🐛 Bug

The current `scheduleReduction` implementation always assumes a non-reduction axis present in the reduction output tensor. This is not the case when reducing to a single scalar value.

## To Reproduce
```python
x = torch.randn([4, 5, 6], dtype=dtype, device=device)
def t(x: torch.Tensor):
  o = torch.add(x, 0)
  o = torch.sum(o, dim=[0,1,2])
  return o
t_jit = torch.jit.script(t)
jit_o = t_jit(x)
jit_o = t_jit(x)
o = t(x)
self.assertEqual(o.dtype, jit_o.dtype)
self.assertEqual(o, jit_o)
self.assertGraphContains(t_jit.graph_for(x), FUSION_GUARD)
```
